build-aux/snap/snapcraft.yaml

```diff
@@ -127,3 +128,26 @@ parts:
       cp -a fc-cache-bionic $SNAPCRAFT_PART_INSTALL/bin/fc-cache-v7
     prime:
       - bin/fc-cache-v7
+  apparmor:
+    plugin: autotools
+    build-packages: [ bison, flex, gettext ]
+    source: https://launchpad.net/apparmor/3.0/3.0.1/+download/apparmor-3.0.1.tar.gz
+    override-build: |
+      cd libraries/libapparmor
+      ./autogen.sh
+      ./configure --prefix=/usr
+      make
+      # place libapparmor into staging area for use by snap-confine
+      make -C src install DESTDIR=$SNAPCRAFT_STAGE
```

Comment:
did you confirm that this is sufficient to make snap-confine use the apparmor from this part? I don't remember all the details of how the deb is built, but since we aren't using a native snapcraft build system, I'm concerned that the deb build for snapd ignores whatever is in $SNAPCRAFT_STAGE (unlike if we were using one of the native snapcraft build plugins, which should pick up stuff in $SNAPCRAFT_STAGE properly).

Reply:
Yes - this is why I added 75c43fc84b5ddff32c9a55ba18577cc2c6bd4f89 which changes configure.ac to look for apparmor-3 specifically when it detects it is building via snapcraft.

Reply:
ack, thanks I just didn't see anything in the configure.ac changes which actually uses the $SNAPCRAFT_STAGE env var, but probably snapcraft sets some other env var that configure.ac uses to find apparmor 3 inside $SNAPCRAFT_STAGE.